exchange.py

```diff
@@ -16,29 +16,47 @@ def __init__(self, configuration):
         self.privatekey = configuration['privatekey']
         self.publickey  = configuration['publickey']
         self.exchange   = configuration['exchange']
-
+        
+        self.DEBUG = 100 # TODO: add to configuration
+        self.api = "https://tuxexchange.com/api" # TODO: add to configuration
+        
+    def debug(self, level, header, *args):
+        if level <= self.DEBUG:
+            print("-"*10)
+            print("{}---> {}:".format(level, header))
+            for i in args:
+                print("\t\t" + repr(i))
+            print("-" * 10)
+
+    def _signed_request(self, param, nonce=None, timeout=15):
```

Comment:
send to utils and rename as sign_request

Reply:
@15chrjef @livinginformation  If I move to utils, I would have to pass in keys as arguments, seems a bit inside out. I think using class attributes would be cleanest, however I'm new to contributing so whatever yall want.

Reply:
I think this is best. IMO the exchange is the owner/source of truth for all state, but it can delegate the responsibility to produce dictionaries of primitives to util functions. 

I prefer to have small classes and files in general, but that is just one way of going about it.